src/Shell/QueueShell.php

```diff
@@ -188,20 +188,20 @@ public function runworker() {
 					$this->_log('job ' . $queuedTask['job_type'] . ', id ' . $queuedTask['id'] . ' failed and requeued', $pid);
 					$this->out('Job did not finish, requeued.');
 				}
-			} elseif (Configure::read('Queue.exitwhennothingtodo')) {
+			} elseif (Configure::readOrFail('Queue.exitwhennothingtodo')) {
```

Comment:
I am a bit afraid that this could fail as patch version too hard, if someone just hasnt set this (really kind of optional) key..

Reply:
I had this doubt as well. You can keep the PR open until you create another major (or *maybe* minor) release. Bugfix is no good 👍

Reply:
We have a new 4.0 coming up, maybe the not so problematic ones can go into there?